mempool/src/core_mempool/transaction_store.rs

```diff
@@ -538,46 +540,54 @@ impl TransactionStore {
     /// Returns block of transactions and new last_timeline_id.
     pub(crate) fn read_timeline(
         &self,
-        timeline_id: u64,
+        timeline_id: &TimelineId,
         count: usize,
-    ) -> (Vec<SignedTransaction>, u64) {
+    ) -> (Vec<SignedTransaction>, TimelineId) {
         let mut batch = vec![];
         let mut batch_total_bytes: u64 = 0;
-        let mut last_timeline_id = timeline_id;
+        let mut last_timeline_id = timeline_id.0.clone();
 
         // Add as many transactions to the batch as possible
-        for (address, sequence_number) in self.timeline_index.read_timeline(timeline_id, count) {
-            if let Some(txn) = self
-                .transactions
-                .get(&address)
-                .and_then(|txns| txns.get(&sequence_number))
-            {
-                let transaction_bytes = txn.txn.raw_txn_bytes_len() as u64;
-                if batch_total_bytes.saturating_add(transaction_bytes) > self.max_batch_bytes {
-                    break; // The batch is full
-                } else {
-                    batch.push(txn.txn.clone());
-                    batch_total_bytes = batch_total_bytes.saturating_add(transaction_bytes);
-                    if let TimelineState::Ready(timeline_id) = txn.timeline_state {
-                        last_timeline_id = timeline_id;
-                    }
-                    if let Ok(time_delta) = SystemTime::now().duration_since(txn.insertion_time) {
-                        counters::core_mempool_txn_commit_latency(
-                            BROADCAST_BATCHED_LABEL,
-                            E2E_LABEL,
-                            time_delta,
-                        );
+        for (i, bucket) in self
+            .timeline_index
+            .read_timeline(timeline_id, count)
+            .iter()
+            .enumerate()
+            .rev()
```

Comment:
it feels fragile to have rev to guarantee the correct ordering, but hopefully test can capture any breakage

Reply:
Added an explicit test for significant ordering